paasta_tools/cli/cmds/spark_run.py

```diff
@@ -537,14 +548,29 @@ def configure_and_run_docker_container(
     system_paasta_config: SystemPaastaConfig,
     spark_conf: Mapping[str, str],
     aws_creds: Tuple[Optional[str], Optional[str], Optional[str]],
+    cluster_manager: str,
 ) -> int:
 
     # driver specific volumes
-    volumes = (
-        spark_conf.get("spark.mesos.executor.docker.volumes", "").split(",")
-        if spark_conf.get("spark.mesos.executor.docker.volumes", "") != ""
-        else []
-    )
+    if cluster_manager == CLUSTER_MANAGER_MESOS:
+        volumes = (
+            spark_conf.get("spark.mesos.executor.docker.volumes", "").split(",")
+            if spark_conf.get("spark.mesos.executor.docker.volumes", "") != ""
+            else []
+        )
+    elif cluster_manager == CLUSTER_MANAGER_K8S:
+        volumes = []
+        volume_names = [
+            re.match('spark.kubernetes.executor.volumes.hostPath.(\d+).mount.path', key).group(1)
+            for key in spark_conf.keys()
```

Comment:
How are spark.kubernetes.executor.volumes.host* being set in spark_conf at first place? Through the env variables populated by the other review?

Reply:
Missed your comment, yes they're set in service-configuration-lib